pkg/argsbuilder/evaluatejob.go

```diff
@@ -0,0 +1,318 @@
+	args        *types.EvaluateJobArgs
+	argValues   map[string]interface{}
+	subBuilders map[string]ArgsBuilder
+}
+
+func NewEvaluateJobArgsBuilder(args *types.EvaluateJobArgs) ArgsBuilder {
+	e := &EvaluateJobArgsBuilder{
+		args:        args,
+		argValues:   map[string]interface{}{},
+		subBuilders: map[string]ArgsBuilder{},
+	}
+
+	e.AddSubBuilder(
+		NewSubmitSyncCodeArgsBuilder(&e.args.SubmitSyncCodeArgs),
+	)
+
+	return e
+}
+
+func (e *EvaluateJobArgsBuilder) GetName() string {
+	items := strings.Split(fmt.Sprintf("%v", reflect.TypeOf(*e)), ".")
+	return items[len(items)-1]
+}
+
+func (e *EvaluateJobArgsBuilder) AddSubBuilder(builders ...ArgsBuilder) ArgsBuilder {
+	for _, b := range builders {
+		e.subBuilders[b.GetName()] = b
+	}
+	return e
+}
+
+func (e *EvaluateJobArgsBuilder) AddArgValue(key string, value interface{}) ArgsBuilder {
+	for name := range e.subBuilders {
+		e.subBuilders[name].AddArgValue(key, value)
+	}
+	e.argValues[key] = value
+	return e
+}
+
+func (e *EvaluateJobArgsBuilder) AddCommandFlags(command *cobra.Command) {
+	for name := range e.subBuilders {
+		e.subBuilders[name].AddCommandFlags(command)
+	}
+
+	var (
+		envs             []string
+		dataDir          []string
+		dataSources      []string
+		annotations      []string
+		labels           []string
+		tolerations      []string
+		nodeSelectors    []string
+		imagePullSecrets []string
+	)
+
+	// evaluate job arguments
+	command.Flags().StringVar(&e.args.Name, "name", "", "the evaluate job name")
+	command.Flags().StringVar(&e.args.Namespace, "namespace", "", "the evaluate job namespace")
+	command.Flags().StringVar(&e.args.ModelName, "model-name", "", "the model name to evaluate")
+	command.Flags().StringVar(&e.args.ModelVersion, "model-version", "", "the model version to evaluate")
+	command.Flags().StringVar(&e.args.ModelPath, "model-path", "", "the model path to evaluate in the container")
+	command.Flags().StringVar(&e.args.DatasetPath, "dataset-path", "", "the model version to evaluate")
+	command.Flags().StringVar(&e.args.MetricsPath, "metrics-path", "", "the evaluate result saved path")
+	command.Flags().StringVar(&e.args.Image, "image", "", "the evaluate image")
+	command.Flags().StringVar(&e.args.WorkingDir, "working-dir", "/root", "working directory to extract the code. If using syncMode, the $workingDir/code contains the code")
+	command.Flags().IntVar(&e.args.GPUCount, "gpus", 0, "the limit GPU count of each replica to run the evaluate job.")
+	command.Flags().StringVar(&e.args.Cpu, "cpu", "", "the request cpu of each replica to run the evaluate job.")
+	command.Flags().StringVar(&e.args.Memory, "memory", "", "the request memory of each replica to run the evaluate job.")
+
+	command.Flags().StringSliceVarP(&envs, "env", "e", []string{}, "the environment variables")
+	command.Flags().StringSliceVarP(&annotations, "annotation", "a", []string{}, "the annotations")
+	command.Flags().StringSliceVarP(&labels, "label", "", []string{}, "the labels")
```

Comment:
please replace StringSliceVarP to StringArrayVarP

Reply:
fixed